blockchain/v2/processor.go

```diff
@@ -87,10 +87,9 @@ func (state *pcState) synced() bool {
 }
 
 func (state *pcState) enqueue(peerID p2p.ID, block *types.Block, height int64) {
-	if _, ok := state.queue[height]; ok {
-		panic("duplicate block enqueued by processor")
+	if _, ok := state.queue[height]; !ok {
+		state.queue[height] = queueItem{block: block, peerID: peerID}
 	}
-	state.queue[height] = queueItem{block: block, peerID: peerID}
 }
```

Comment:
Since the old code panics, it seems to me like it's meant to have an invariant that a block is only enqueued once. Instead of removing the check, maybe we need to investigate why the invariant is broken? @brapse may have some input.

Also, what happens if we receive the block from the old peer after enqueueing the block with a new peer?

Reply:
I got the impression it's possible for reactor to ask 2 or more peers for the same block. Therefore, we might receive the same block twice, which will lead to the above panic.

Reply:
It may be possible (since we have the panic), but I don't think it's meant to be possible. The scheduler explicitly allocates a height to a single peer, and errors out if it receives a block from a different peer:

https://github.com/tendermint/tendermint/blob/346aa14db560c3c867a3e00b4fbd8b000dcffe30/blockchain/v2/scheduler.go#L361-L363

Reply:
maybe it happens when current peer sends two identical blocks. I just wrote a test, which leads to panic